reuse.go

```diff
@@ -110,14 +111,34 @@ func (r *reuseBase) runGarbageCollector() {
 }
 
 // must be called while holding the mutex
-func (r *reuseBase) maybeStartGarbageCollector() {
+func (r *reuse) maybeStartGarbageCollector() {
 	if !r.garbageCollectorRunning {
 		r.garbageCollectorRunning = true
 		go r.runGarbageCollector()
 	}
 }
+func (r *reuse) Dial(network string, raddr *net.UDPAddr) (*reuseConn, error) {
+	ips := []net.IP{}
+	if router, err := netroute.New(); err == nil {
+		_, _, src, err := router.Route(raddr.IP)
```

Comment:
Do we need to handle `err != nil` here?

Reply:
there isn't a great `nil` value for src - checking for nil error seems like a reasonable sanity check before validating that the source hasn't been left as all 0's.